common/federation/src/main/java/app/coronawarn/server/common/federation/client/CloudFederationFeignHttpClientProvider.java

```diff
@@ -61,15 +67,15 @@ private ApacheHttpClientFactory federationHttpClientFactory(int connectionPoolSi
         .setMaxConnPerRoute(connectionPoolSize)
         .setMaxConnTotal(connectionPoolSize)
         .setSSLContext(getSslContext(keyStorePath, keyStorePass))
-        .setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE));
+        .setSSLHostnameVerifier(this.hostnameVerifierProvider.createHostnameVerifier()));
   }
 
   private SSLContext getSslContext(File keyStorePath, String keyStorePass) {
     try {
       return SSLContextBuilder
           .create()
           .loadKeyMaterial(keyStorePath, keyStorePass.toCharArray(), keyStorePass.toCharArray())
-          .loadTrustMaterial(TrustSelfSignedStrategy.INSTANCE)
+          .loadTrustMaterial(this.keyTruststore, this.keyTruststorePassword.toCharArray())
```

Comment:
The problem with this is that now running Upload locally (using `fake-client` profile) will fail. Because we don't have a keystore for that. 

Also, Upload and Download will fail in dev because we don't have a keystore there either.